src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java

```diff
@@ -226,19 +226,30 @@ private void executeColumnInsert(Connection conn, ArrayList<Object> columnData)
 		columnData.clear();
 	}
 
-	public static MysqlSavedSchema restore(Connection connection, MaxwellContext context) throws SQLException, IOException, InvalidSchemaError {
-		MysqlSavedSchema s = new MysqlSavedSchema(context.getServerID(), context.getCaseSensitivity());
+	public static MysqlSavedSchema restore(MaxwellContext context, BinlogPosition targetPosition) throws SQLException, InvalidSchemaError {
+		try ( Connection conn = context.getMaxwellConnection() ) {
+			Long schemaID = findSchema(conn, targetPosition, context.getServerID());
+			if (schemaID == null)
+				return null;
```

Comment:
Before it'd raise... so I suppose MysqlSchemaStore will just capture a new schema for the same binlong initial position? does `null` here means maxwell is in a bad state (saved position but missing schema) or does it mean that potentially maxwell was down for too long, and updates were lost?